packages/cadl-ranch-specs/http/type/union/main.tsp

```diff
@@ -3,139 +3,199 @@ import "@azure-tools/cadl-ranch-expect";
 
 using TypeSpec.Http;
 
-@doc("Test for type of union.")
+/**
+ * Describe scenarios for various combinations of unions.
+ */
 @supportedBy("dpg")
 @scenarioService("/type/union")
 namespace Type.Union;
 
-@doc("This is a base model.")
-model BaseModel {
+model Cat {
   name: string;
 }
 
-@doc("The first one of the unioned model type.")
-model Model1 extends BaseModel {
-  prop1: int32;
+model Dog {
+  bark: string;
 }
 
-@doc("The second one of the unioned model type.")
-model Model2 extends BaseModel {
-  prop2: int32;
+enum LR {
+  left,
+  right,
 }
-
-union MyNamedUnion {
-  one: Model1,
-  two: Model2,
+enum UD {
+  up,
+  down,
 }
 
-model ModelWithSimpleUnionProperty {
-  simpleUnion: int32 | int32[];
+/**
+ * Describe union of string "a" | "b" | "c"
+ */
+@route("/strings-only")
+interface StringsOnly extends GetAndSend<"a" | "b" | "c", "\"b\""> {}
+
+/**
+ * Describe union of string string | "b" | "c"
+ */
+@route("/string-extensible")
+interface StringExtensible extends GetAndSend<string | "b" | "c", "\"custom\""> {}
+
+union StringExtensibleNamedUnion {
+  string,
+  OptionB: "b",
+  "c",
 }
-
-model ModelWithNamedUnionProperty {
-  namedUnion: MyNamedUnion;
+/**
+ * Describe union of string string | "b" | "c" but where the union is named and some of the variants are named
+ */
+@route("/string-extensible-named")
+interface StringExtensibleNamed extends GetAndSend<StringExtensibleNamedUnion, "\"custom\""> {}
+
+/**
+ * Describe union of integer 1 | 2 | 3
+ */
+@route("/ints-only")
+interface IntsOnly extends GetAndSend<1 | 2 | 3, "2"> {}
+
+/**
+ * Describe union of floats 1.1 | 2.2 | 3.3
+ */
+@route("/floats-only")
+interface FloatsOnly extends GetAndSend<1.1 | 2.2 | 3.3, "2.2"> {}
+
+/**
+ * Describe union of models
+ */
+@route("/models-only")
+interface ModelsOnly extends GetAndSend<Cat | Dog, """
+  {
+    "name": "test"
+  }
+  """> {}
+
+model EnumsOnlyCases {
+  /** This should be receive/send the left variant */
+  lr: LR | UD;
+
+  /** This should be receive/send the up variant */
+  ud: UD | UD;
 }
 
-model ModelWithSimpleUnionPropertyInResponse {
-  simpleUnion: string | int32[];
+/**
+ * Describe union of 2 different enums
+ */
+@route("/enums-only")
+interface EnumsOnly extends GetAndSend<LR | UD, """
+  {
+    "lr": "right",
+    "ud": "up"
+  }
+  """, EnumsOnlyCases> {}
+
+model StringAndArrayCases {
+  /** This should be receive/send the string variant */
+  string: string;
+
+  /** This should be receive/send the array variant */
+  array: string[];
 }
 
-model ModelWithNamedUnionPropertyInResponse {
-  namedUnion: MyNamedUnion;
+/**
+ * Describe union of a string and an array of strings
+ */
+@route("/string-and-array")
+interface StringAndArray
+  extends GetAndSend<
+      string | string[],
+      """
+      {
+        "string": "test",
+        "array": ["test1", "test2"]
+      }
+      """,
+      StringAndArrayCases
+    > {}
+
+alias MixedTypesUnion = Cat | "a" | 2 | 3.3 | true;
```

Comment:
Emm, so this is a mix of model and literal (not model and scala).

Reply:
Right maybe can split into mixed literals and mixed types

Reply:
Yeah, maybe we can just insert a scala to this MixedTypesUnion as well :-)

Reply:
Though Java likely going to have a problem with NumericLiteral. It is hard to tell whether the literal is an integer or float. <-- that is to be solved within Java, maybe always use float.

Reply:
yeah though I can see emitters being able to do a better job if its a union of literals vs a union of basically anything(which approach just putting `unknown`)